### PR TITLE
[Front] Address/Postal Code/City support on Position GUI (#2843)

### PR DESCRIPTION
**Pull request by Archidoit**
### PR content
- adding street_address and postal_code in the Position edition form
- adding street_address and postal_code in the Position creation form
- adding a 'Details' tab in Position>Overview
- adding missing translations for Position
- fixing Positions lines blinking (migrations to tsx)

### Related issues

https://github.com/OpenCTI-Platform/opencti/issues/2843

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/locations/Positions.js [hunk 1]
```diff
@@ -1,191 +0,0 @@
...
-    const params = buildViewParamsFromUrlAndStorage(
-      props.history,
-      props.location,
-      'view-positions',
-    );
-    this.state = {
-      sortBy: R.propOr('name', 'sortBy', params),
-      orderAsc: R.propOr(true, 'orderAsc', params),
-      searchTerm: R.propOr('', 'searchTerm', params),
-      view: R.propOr('lines', 'view', params),
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/locations/Positions.js:33]: The view was not used right ? This is why you remove it.
I don't see it on the master.

File: opencti-platform/opencti-front/src/private/components/locations/positions/PositionDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+interface PositionDetailsProps {
+  position: Position_position$data
+}
+
+const PositionDetails: FunctionComponent<PositionDetailsProps> = ({ position }) => {
+  const { t } = useFormatter();
+  const classes = useStyles();
+
+  return (
+    <QueryRenderer
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/locations/positions/PositionDetails.tsx]: It's will be better to use a suspense.

File: opencti-platform/opencti-front/src/private/components/locations/positions/PositionDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        positionDetailsLocationRelationshipsLinesQuery
+      }
+      variables={{
+        count: 20,
+        elementId: [position.id],
+        relationship_type: 'located-at',
+      }}
+      render={({ props }: { props: PositionDetailsLocationRelationshipsLinesQueryLinesPaginationQuery$data }) => {
+        const targets = props?.stixCoreRelationships?.edges;
+        const cities = targets?.filter((n) => n?.node.to?.entity_type === 'City').map((n) => n?.node.to?.name);
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/locations/positions/PositionDetails.tsx]: If we have a relationship, the node, to, to.id and to.entity_type property should not be null, what do you think ?

File: opencti-platform/opencti-front/src/private/components/locations/positions/PositionEditionOverview.js [hunk 4]
```diff
@@ -318,6 +322,32 @@ class PositionEditionOverviewComponent extends Component {
                 <SubscriptionFocus context={context} fieldName="longitude" />
               }
             />
+            <Field
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/locations/positions/PositionEditionOverview.js:325]: We don't want this fields in creation as well ?

File: opencti-platform/opencti-front/src/private/components/locations/Positions.js (hunk 1 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/locations/Positions.js:33]: It was used but was always equal to 'lines' and it's value was never changed. There were no behavior for view != 'lines'. So it was not necessary to keep it.

**Comment by RomuDeuxfois** [comment]
Tested : OK

We have no limit on the display of country, region, ect...
I don't know if it's possible to have a lot of them, but it can be a problem
image

--> the number of location to display is limited to 20 by security (even if it should never reach 20 in theory)

**Comment by RomuDeuxfois** [comment]
Bug founds : I can't creation a Position right now

**Review by Kedae — APPROVED** [review]

**Pull request merged**